### ISSUE TITLE
What groups are there for joshgone commits?

### ISSUE THREAD

**Issue by leonsilicon** [GeeTransit/joshgone#2]
So, I know you're starting to use conventional commits (e.g. feat(chants) or docs(music), or feat(*)), and I'm wondering what are the possible groups for authoring conventional commits for the joshgone repo (i.e. what are all the possible strings that can be substituted in `optional scope` of `<type>[optional scope]: <description>` (from https://www.conventionalcommits.org/en/v1.0.0-beta.4/). 

Also, what does the * mean in the scope of conventional commits (e.g. feat(*))?

Also, is it fine to use the breaking change syntax for the conventional commit syntax (https://www.conventionalcommits.org/en/v1.0.0-beta.4/#commit-message-with-optional--to-draw-attention-to-breaking-change)? Is this syntax adhered to in the conventional commits for the joshgone repo?

Also, are you planning to use conventional commits for autogenerating changelogs (https://www.conventionalcommits.org/en/v1.0.0-beta.4/#why-use-conventional-commits)? And if so, do you have some system where if it detects a commit with a breaking change (i.e. prefixed with `!`), it will automatically bump the major version number? If so, do you mind sharing that tool (because that sounds pretty useful).

Also, what tool are you using to automatically generate the changelogs from your commits? Is it language-agnostic?

Speaking about bumping version numbers based on commits, it looks like the conventional commits specification encourages bumping the version number based on what type of commits there are: https://www.conventionalcommits.org/en/v1.0.0-beta.4/#how-does-this-relate-to-semver. Would this be adhered to in the joshgone repo?

Also, if I were to contribute to this repo, would I have to make sure all my commits before the actual merge follow the convention? It looks like based on this link https://www.conventionalcommits.org/en/v1.0.0-beta.4/#how-does-this-relate-to-semver that you would be able to squash all these commits.

Also, I'm looking at your recent commits, and it looks like they don't really have a scope:
<img width="617" alt="image" src="https://user-images.githubusercontent.com/36966635/157474449-74c551ae-47e4-43fe-ac60-fa86f171a7fb.png">

Would this be fine for me to do when I'm contributing or would you prefer that I always add a scope?

Thanks for reading this longue issue!

**Comment by GeeTransit** [comment]
> what are the possible groups for authoring conventional commits

Currently groups are optional but I'm expecting each cog to be its own group (like chants, music, ...), though this isn't enforced (as I'm still testing the waters for conventional commits).

> what does the * mean in the scope of conventional commits (e.g. feat(*))?

I'm not exactly sure either, but I suspect its a catch-all for automated changelog generators to force inclusion or something.

**Comment by GeeTransit** [comment]
(whoops I pressed Ctrl+Enter by accident)

> ... is it fine to use the breaking change syntax for the conventional commit syntax?

Yep, though ideally there should be no breaking changes without a deprecation period.

> ... do you have some system where ... a breaking change ... will automatically bump the major version number?

Unfortunately I don't think I've found a tool that does that (though maybe there's one for the Node.js community). While searching, I skipped over tools that required SemVer ([see this blog for my reasoning](https://snarky.ca/why-i-dont-like-semver/#version-numbers-are-your-branching-strategy-as-a-sequence-of-numbers)) so there's probably ones that _do_ work if SemVer is your versioning strategy.

> ... what tool are you using to automatically generate the changelogs from your commits?

I'm testing out [git-chglog](https://github.com/git-chglog/git-chglog) currently. It generates a Markdown changelog based on the commit history which is pretty neat, though it seems it might be dead :( I'm also looking into [sphinx-github-changelog](https://github.com/ewjoachim/sphinx-github-changelog) for generating changelogs, though I'm not sure about its support for out of order releases (not that it'll matter though).

I'm also deciding between generating from the Git commit history or from GitHub Pull Requests (via [autogenerated release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes)). I'm slightly leaning towards the latter right now. A concern I had with using GitHub Releases was vendor lock-in, though I'm also planning on using GitHub Actions to automatically generate & publish documentation to GitHub Pages and to build & publish new releases to PyPI, so it's not that big of a concern.

> Would [bumping the version number based on what type of commits there are] be adhered to in the joshgone repo?

Joshgone doesn't really have a version (though I guess there could be one if you count the output of `hatch version` to be one). I don't see a reason for strictly defined versions for the foreseeable future. If I do end up generating changelogs, versions are probably gonna be of the form v0.X and be incremented every once in a while (after major changes are merged).

> ... would I have to make sure all my commits before the actual merge follow the convention?

I'll probably squash them before merging.

> ... would you prefer that I always add a scope?

They can be scope-less for now :)
